lib/checkcondition.cpp

```diff
@@ -1188,12 +1187,13 @@ void CheckCondition::alwaysTrueFalse()
                 continue;
             if (Token::Match(tok, "! %num%|%bool%"))
                 continue;
+            if (Token::Match(tok, "&&|%oror%") && !Token::Match(tok->astOperand1(), "%char%") && !Token::Match(tok->astOperand2(), "%char%"))
```

Comment:
I see no test case where there is warning for `&&` or `||` .. but I don't see why we would want to have it.

Reply:
That is to fix [this test case](https://github.com/danmar/cppcheck/blob/master/test/testcondition.cpp#L2431) with a char. Let me check if this is still necessary.

Reply:
The test case still fails when removing the extra check.